src/mtype.c

```diff
@@ -6516,10 +6516,23 @@ void TypeQualified::resolveHelper(Loc loc, Scope *sc,
             // If the symbol is an import, try looking inside the import
             if (Import *si = s->isImport())
             {
-                s = si->search(loc, s->ident);
-                if (s && s != si)
-                    goto L1;
-                s = si;
+                if (idents.dim == 0)
+                {
+                    /* Special case for runnable/Same.d:
```

Comment:
Yes, (maybe) it's very weird feature for the symbols parsed as types. I didn't know about that when I noticed.